### PR TITLE
feat: add progress bar support for image rescan (#898)

### PR DESCRIPTION
**Pull request by mickowale**
(#898)

### CODE REVIEW

File: src/lib/backend.ai-client-node.js [hunk 1]
```diff
@@ -2428,6 +2428,11 @@ class Maintenance {
         this.client = client;
         this.urlPrefix = '/resource';
     }
+    attach_background_task(task_id) {
```
Comment (inureyes) [inline src/lib/backend.ai-client-node.js:2435]: @mickowale We need a comment section above this method. (It is referenced by TypeScript to check the parameter type)

File: src/lib/backend.ai-client-node.ts [hunk 2]
```diff
@@ -2633,6 +2633,12 @@ class Maintenance {
     this.urlPrefix = '/resource';
   }
 
+  attach_background_task(task_id: string) {
```
Comment (inureyes) [inline src/lib/backend.ai-client-node.ts:2640]: @mickowale We need a comment section above this method. (It is referenced by TypeScript to check the parameter type)

**Review by inureyes — CHANGES_REQUESTED** [review]
@mickowale Please add comment to the methods. Other things looks fine (need more test / migrate into sidebar but it will be handled with another issue)

**Comment by adrysn** [comment]
There is also a rescan images in backend-ai-registry-list.ts to update individual registries. I think it needs to be updated as well.

**Comment by mickowale** [comment]
> There is also a rescan images in backend-ai-registry-list.ts to update individual registries. I think it needs to be updated as well.

I have updated this now and also merged with the main branch. I will squash the commits if the current state is okay.

**Review by inureyes — APPROVED** [review]
LGTM.

**Comment by inureyes** [comment]
@mickowale Good! I'll squash and merge into main branch. :D

**Pull request merged**